api/app/src/main/kotlin/packit/service/RolePermissionService.kt

```diff
@@ -0,0 +1,64 @@
+package packit.service
+
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import packit.exceptions.PackitException
+import packit.model.Role
+import packit.model.RolePermission
+import packit.model.dto.UpdateRolePermission
+import packit.repository.*
+
+interface RolePermissionService
+{
+    fun getRolePermissionsToUpdate(role: Role, addRolePermissions: List<UpdateRolePermission>): List<RolePermission>
+    fun removeRolePermissionsFromRole(role: Role, removeRolePermissions: List<UpdateRolePermission>)
+}
+
+@Service
+class BaseRolePermissionService(
+    private val permissionRepository: PermissionRepository,
+    private val packetRepository: PacketRepository,
+    private val packetGroupRepository: PacketGroupRepository,
+    private val tagRepository: TagRepository,
+    private val rolePermissionRepository: RolePermissionRepository
+) : RolePermissionService
+{
+    override fun getRolePermissionsToUpdate(
+        role: Role,
+        addRolePermissions: List<UpdateRolePermission>
```

Comment:
```suggestion
        updateRolePermissions: List<UpdateRolePermission>
```

Reply:
i named like that to differentiate the adding and deleting of role permissions

Reply:
But you're using it for permissions being removed too: 
https://github.com/mrc-ide/packit/pull/64/files#diff-91662b556b4c3ee7dc5037863b1a55465c6b8218de7a6675c404baec77f0fb66R56

Reply:
oh yup was mistaken have updated in PR --> https://github.com/mrc-ide/packit/pull/68